blockstore/src/main/java/brooklyn/location/blockstore/NewVolumeCustomizer.java

```diff
@@ -3,24 +3,51 @@
 import brooklyn.location.blockstore.api.MountedBlockDevice;
 import brooklyn.location.blockstore.api.VolumeManager;
 import com.google.common.base.Optional;
+import com.google.common.reflect.TypeToken;
+import org.apache.brooklyn.config.ConfigKey;
+import org.apache.brooklyn.core.config.ConfigKeys;
 import org.apache.brooklyn.location.jclouds.BasicJcloudsLocationCustomizer;
+import org.apache.brooklyn.location.jclouds.JcloudsLocation;
 import org.apache.brooklyn.location.jclouds.JcloudsMachineLocation;
+import org.apache.brooklyn.util.collections.MutableList;
+import org.apache.brooklyn.util.collections.MutableMap;
+import org.apache.brooklyn.util.core.config.ConfigBag;
+import org.jclouds.compute.ComputeService;
 import org.jclouds.compute.domain.NodeMetadata;
 
+import java.util.List;
 import java.util.Map;
 
 public abstract class NewVolumeCustomizer extends BasicJcloudsLocationCustomizer {
 
-    public NewVolumeCustomizer(Map<BlockDeviceOptions, FilesystemOptions> volumes) {
-        this.volumes = volumes;
-        this.mountedBlockDevice = null;
+    public static final ConfigKey<List<Map<BlockDeviceOptions, FilesystemOptions>>> VOLUMES = ConfigKeys.newConfigKey(
```

Comment:
How are you going to set `BlockDeviceOptions` & `FilesystemOptions` from YAML? Better change to `List<Map<?, ?>>`